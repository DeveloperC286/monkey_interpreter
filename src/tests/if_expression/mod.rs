use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case(
        "if (10 > 1) { if (10 > 1) { return TRUE;} return FALSE;}",
        "if_expression_case_1"
    ),
    case("if (10 > 1) { return 1 + 5; } return 1;", "if_expression_case_2"),
    case("if (1) { 1 }", "if_expression_case_3"),
    case("if (1 < 2) { 4 } else { true }", "if_expression_case_4"),
    case("if (true) { 10 }", "if_expression_case_5"),
    case("if (false) { 10 }", "if_expression_case_6"),
    case("if (1 > 2) { 5 }", "if_expression_case_7"),
    case("if (4 > 6) { true } else { false }", "if_expression_case_8"),
    case("if (false) { 85 } else { 58 }", "if_expression_case_9")
)]
fn if_expression_cases(code: &str, snapshot_name: &str) {}

#[apply(if_expression_cases)]
fn test_if_expression_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(if_expression_cases)]
fn test_if_expression_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(if_expression_cases)]
fn test_if_expression_evaluation(code: &str, snapshot_name: &str) {
    assert_evaluation!(code, snapshot_name);
}

#[apply(if_expression_cases)]
fn test_if_expression_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
