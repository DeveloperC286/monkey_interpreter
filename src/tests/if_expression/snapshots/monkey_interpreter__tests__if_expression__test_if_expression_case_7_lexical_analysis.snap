---
source: src/tests/if_expression/mod.rs
expression: tokens
---
[
    If,
    OpeningRoundBracket,
    Integer {
        literal: 1,
    },
    GreaterThan,
    Integer {
        literal: 2,
    },
    ClosingRoundBracket,
    OpeningCurlyBracket,
    Integer {
        literal: 5,
    },
    ClosingCurlyBracket,
    EndOfFile,
]
